### PR TITLE
Use Published Dependencies For Chisel + FIRRTL

### PR DESCRIPTION
**Pull request by davidbiancolin**
Building Chisel + FIRRTL + friends from source is becoming untenable due to enhanced compiler plugin support amoungst other things.  Since these tools are much more stable than they used to be the need to build specific versions from source is no longer as strong. Devs can still rely on publish-local should they wish to deliver a custom version of firrtl or chisel. 

Since firrtl.testutils is not published, i've elected to copy the sources into FireSim instead of building a jar from source. 

#### Related PRs / Issues

This is the FIreSim-side PR for ucb-bar/chipyard#1054

#### UI / API Impact

Should not affect users, will make development slightly more difficult when the dev wants to hack on FIRRTL / Chisel sources.

#### Verilog / AGFI Compatibility

Will regenerate AGFIs.

### Contributor Checklist
- [x] Did you set dev as the base branch?
- [x] Did you add Scaladoc to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous prints/debugging code?
- [x] Did you state the UI / API impact?
- [x] Did you specify the Verilog / AGFI compatibility impact?

- [x] (If applicable) Did you regenerate and publicly share default AGFIs?

- [ ] (If applicable) Did you mark the PR as "Please Backport"?
- [ ] (On merge) Did you update release notes in the dev-to-master PR ?

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you mark the proper release milestone?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?

### CODE REVIEW

File: sim/Makefile [hunk 1]
```diff
@@ -47,11 +47,8 @@ SCALA_BUILDTOOL_DEPS ?= $(sbt_sources)
 
 SBT_THIN_CLIENT_TIMESTAMP = $(base_dir)/project/target/active.json
 
-# By default build chisel3/firrtl and other subprojects from source
-# Workaround: Specify a firrtl version in system properties so that Treadle uses a
-# compatible version of FIRRTL and not 1.5-SNAPSHOT (which is the default
-# specified in it's build.sbt, and is not overridden by Chipyard's build.sbt)
-override SBT_OPTS += -DfirrtlVersion=1.4.1 -Dsbt.sourcemode=true -Dsbt.workspace=$(chipyard_dir)/tools
+# by default build chisel3/firrtl and other subprojects from source
+override SBT_OPTS += -Dsbt.sourcemode=true -Dsbt.workspace=$(chipyard_dir)/tools
```
Comment (abejgonzalez) [inline sim/Makefile:51]: Why not remove sriracha like Chipyard?

File: sim/midas/src/main/scala/midas/passes/fame/FAMETransform.scala [hunk 2]
```diff
@@ -223,7 +223,7 @@ object FAMEModuleTransformer {
           XDCFiles.Implementation,
           s"""|create_generated_clock -name ${clockName} -source [get_pins -of [get_clocks host_clock]] [get_pins {}] -divide_by 1
               |set_multicycle_path $clockMFMR -setup -from [get_clocks $clockName] -to [get_clocks $clockName]
-              |set_multicycle_path 1 -hold  -from [get_clocks $clockName] -to [get_clocks $clockName]
+              |set_multicycle_path ${clockMFMR - 1} -hold  -from [get_clocks $clockName] -to [get_clocks $clockName]
```
Comment (abejgonzalez) [inline sim/midas/src/main/scala/midas/passes/fame/FAMETransform.scala:226]: Should this be put in another PR? Or can you explain this in the PR notes.

File: sim/build.sbt [hunk 3]
```diff
@@ -41,47 +42,35 @@ lazy val chipyardDir = if(firesimAsLibrary) {
 }
 
 lazy val chipyard      = ProjectRef(chipyardDir, "chipyard")
-lazy val chisel        = ProjectRef(workspaceDirectory / "chisel3", "chisel")
-lazy val firrtl        = ProjectRef(workspaceDirectory / "firrtl", "firrtl")
```
Comment (abejgonzalez) [inline sim/build.sbt:45]: Just to double-check... FIRRTL was only needed for some test sources, which are now copied into FireSim?

File: sim/build.sbt [hunk 4]
```diff
@@ -1,6 +1,5 @@
 import Tests._
 
-// This needs to stay in sync with the chisel3 and firrtl git submodules
 val chiselVersion = "3.4.1"
```
Comment (abejgonzalez) [inline sim/build.sbt:3]: Similar to CY's comment. Can we confirm what Chisel version is really used in the final jar?

**Review by abejgonzalez — COMMENTED** [review]
Overall seems good to me!

File: sim/Makefile (hunk 1 above)
Comment (davidbiancolin) [reply sim/Makefile:51]: oops. Will do.

File: sim/midas/src/main/scala/midas/passes/fame/FAMETransform.scala (hunk 2 above)
Comment (davidbiancolin) [reply sim/midas/src/main/scala/midas/passes/fame/FAMETransform.scala:226]: This was necessary to regenerate AGFIs. I'll update the PR notes.

File: sim/build.sbt (hunk 4 above)
Comment (davidbiancolin) [reply sim/build.sbt]: 3.4.1

File: sim/build.sbt (hunk 3 above)
Comment (davidbiancolin) [reply sim/build.sbt:45]: The dependencies on classes in `firrtl % main` is brought it transitively through chisel. Classes in `test` are not published regardless, hence the copied sources.

**Review by abejgonzalez — APPROVED** [review]
LGTM.
